src/main/java/me/magnet/consultant/ServiceIdentifier.java

```diff
@@ -123,6 +138,7 @@ public String toString() {
 		getDatacenter().ifPresent(dc -> descriptors.add("dc=" + dc));
 		getHostName().ifPresent(host -> descriptors.add("host=" + host));
 		getInstance().ifPresent(instance -> descriptors.add("instance=" + instance));
+		getTags().stream().forEach((tag) -> descriptors.add("tag=" + tag));
```

Comment:
In that case (which I'm fine with) I'm opting for a result similar to `oauth[dc=eu-central,host=docker0,instance=<uuid>,tags=[192,staging,testing]]`. This makes it clearer it represents a set. Additionally this is a lot clearer when dealing with an empty set of tags.